### PR TITLE
FEATURE: Tenor as a provider

### PR DESCRIPTION
**Pull request by merefield**
Adds Tenor as a provider option alongside Giphy.

Please note currently the variable names have been updated to make them more generic. This will break existing installs who will have to re-apply their key.

I can modify this on request so that existing installs don't break, but this will compromise the naming convention. I'll discuss with Team and follow up.

NB: this was delivered as part of a commercial arrangement.  My client is currently testing but don't anticipate any issues with sign-off.

### CODE REVIEW

**Review by pmusaraj — CHANGES_REQUESTED** [review]
This looks nice! 

I would avoid changing the setting names as much as possible though. I could live with `giphy_locale` being renamed to `gifs_locale`, but the others don't need to be renamed IMO. It's too much disruption for existing component users and we don't have a way to migrate these values at all.

Curious what @xfalcox thinks.

**Comment by merefield** [comment]
@pmusaraj thanks for your feedback.  Have added you to the Meta DM with @xfalcox

**Comment by merefield** [comment]
@pmusaraj, @xfalcox, thanks for your review, changes completed, all existing settings names persist and i've cleaned up the tenor ones to follow same format.

File: javascripts/discourse/controllers/gif.js.es6 [hunk 1]
```diff
@@ -50,7 +56,10 @@ export default Controller.extend(ModalFunctionality, {
 
       $.ajax({ url: this.getEndpoint(this.query, this.offset) })
         .done((response) => {
-          const images = response.data.map((gif) => ({
+          let images;
```
Comment (pmusaraj) [inline javascripts/discourse/controllers/gif.js.es6:59]: Nitpick: indenting seems off in this section.

**Review by xfalcox — APPROVED** [review]

**Comment by xfalcox** [comment]
Looking great @merefield! I will merge it tomorrow.

File: settings.yml [hunk 2]
```diff
@@ -1,11 +1,26 @@
...
+  type: string
+  default: ""
+  description:
+    en: "TENOR API key"
 giphy_locale:
   type: enum
   default: en
-  description: "Language to use on the search. Used to cater search to regional content. Set to your default forum locale."
+  description:
+    en: "Language to use on the search. Used to cater search to regional content. Set to your default forum locale."
```
Comment (pmusaraj) [inline settings.yml:23]:
```suggestion
    en: "GIPHY: Language to use for search results. Used to cater search to regional content. Set to your default forum locale."
```

File: settings.yml [hunk 3]
```diff
@@ -42,17 +57,38 @@ giphy_locale:
 giphy_file_format:
   type: enum
   default: webp
-  description: "Image format to use for Giphys. WEBP has smaller files that load quicker while GIF provides compatibility with old browsers."
+  description:
+    en: "GIPHY: Image format to use for Giphys. WEBP has smaller files that load quicker while GIF provides compatibility with old browsers."
```
Comment (pmusaraj) [inline settings.yml:61]:
```suggestion
    en: "GIPHY: Image format to use. WEBP has smaller files that load quicker while GIF provides compatibility with old browsers."
```

File: settings.yml [hunk 4]
```diff
@@ -42,17 +57,38 @@ giphy_locale:
...
+    - mediumgif
+    - tinygif
+    - nanogif
 giphy_content_rating:
   type: enum
   default: r
-  description: |
-    Content rating for gif search results, more info at <a href="https://developers.giphy.com/docs/optional-settings#rating">https://developers.giphy.com/docs/optional-settings#rating</a>
+  description:
+    en: 'GIPHY: Content rating for Giphy search results, more info at <a href="https://developers.giphy.com/docs/optional-settings#rating">https://developers.giphy.com/docs/optional-settings#rating</a>'
```
Comment (pmusaraj) [inline settings.yml:79]:
```suggestion
    en: 'GIPHY: Content rating for search results, more info at <a href="https://developers.giphy.com/docs/optional-settings#rating">https://developers.giphy.com/docs/optional-settings#rating</a>.'
```

**Review by pmusaraj — APPROVED** [review]
Looks good to me, thanks for the changes. Made a few small suggestions.

File: settings.yml (hunk 2 above)
Comment (merefield) [reply settings.yml:23]: This is generic to both Tenor and Giphy so the description should remain generic.  The setting is only called giphy_locale to provide upgrade compatibility.

File: javascripts/discourse/controllers/gif.js.es6 (hunk 1 above)
Comment (merefield) [reply javascripts/discourse/controllers/gif.js.es6:59]: Thanks, apologies, I pushed an additional commit.

**Pull request merged**